ios/PurchasesHybridCommon/PurchasesHybridCommon/RCCommonFunctionality.m

```diff
@@ -551,38 +559,39 @@ + (void)productNotFoundErrorWithDescription:(NSString *)errorDescription
 #pragma mark - helpers
 
 + (void)productWithIdentifier:(NSString *)productIdentifier
-              completionBlock:(void (^)(SKProduct * _Nullable))completion {
-    [RCPurchases.sharedPurchases productsWithIdentifiers:@[productIdentifier]
-                                         completionBlock:^(NSArray<SKProduct *> *_Nonnull products) {
-                                             SKProduct *aProduct = nil;
-                                             for (SKProduct *p in products) {
-                                                 if ([productIdentifier isEqualToString:p.productIdentifier]) {
-                                                     aProduct = p;
-                                                 }
-                                             }
-                                             completion(aProduct);
-                                         }];
+              completionBlock:(void (^)(RCStoreProduct * _Nullable))completion {
+    [RCPurchases.sharedPurchases getProductsWithIdentifiers:@[productIdentifier]
+                                                 completion:^(NSArray<RCStoreProduct *> *_Nonnull products) {
+                                                     RCStoreProduct *aProduct = nil;
+                                                     for (RCStoreProduct *p in products) {
```

Comment:
Isn't this unnecessary? It will never return a product with different IDs. I'd just call the completion block with the one product in the array.

Reply:
Ummm... I think you're right! I have no idea why this was in there but it does seem like it's not necessary 😊